FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download/DemoTasks/S3DownloadHTTPExample.c

```diff
@@ -750,25 +753,45 @@ static BaseType_t prvDownloadS3ObjectFile( const TransportInterface_t * pxTransp
 
         if( xHTTPStatus == HTTPSuccess )
         {
-            LogDebug( ( "Received HTTP response from %s%s...",
-                        cServerHost, pcPath ) );
-            LogDebug( ( "Response Headers:\n%.*s",
-                        ( int32_t ) xResponse.headersLen,
-                        xResponse.pHeaders ) );
-            LogInfo( ( "Response Body:\n%.*s\n",
-                       ( int32_t ) xResponse.bodyLen,
-                       xResponse.pBody ) );
-
-            /* We increment by the content length because the server may not
-             * have sent us the range we request. */
-            xCurByte += xResponse.contentLength;
-
-            if( ( xFileSize - xCurByte ) < xNumReqBytes )
+            /* Check for the "Connection: close" header in case the connection
+             * needs to be re-established. */
+            if( xResponse.respFlags == HTTP_RESPONSE_CONNECTION_CLOSE_FLAG )
             {
-                xNumReqBytes = xFileSize - xCurByte;
+                LogInfo( ( "'Connection:close' header found in server response. Attempting to re-connect to the server." ) );
+
+                /* Disconnect and re-establish the connection. */
+                TLS_FreeRTOS_Disconnect( &xNetworkContext );
+                xStatus = connectToServerWithBackoffRetries( prvConnectToServer,
+                                                             &xNetworkContext );
+
+                if( xStatus != pdPASS )
+                {
+                    LogError( ( "Could not reconnect to server. Exiting loop." ) );
+                    break;
+                }
+            }
+            else
```

Comment:
If the connection: close header is found, we still want to do the stuff in this else-case. The response will still contain the partial content.